File: code/src-tauri/src/models/ai.rs
use serde::{Deserialize, Serialize};

/// AI 提供商
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Default)]
#[serde(rename_all = "lowercase")]
pub enum AIProvider {
    #[default]
    OpenAI,
    Claude,
    Ollama,
    Custom,
}

/// AI 配置
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AIConfig {
    pub provider: AIProvider,
    pub api_key: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub base_url: Option<String>,
    pub model: String,
    pub language: String,
    pub auto_review: bool,
}

impl Default for AIConfig {
    fn default() -> Self {
        Self {
            provider: AIProvider::OpenAI,
            api_key: String::new(),
            base_url: None,
            model: "gpt-4o-mini".to_string(),
            language: "zh".to_string(),
            auto_review: false,
        }
    }
}

/// 问题严重程度
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "lowercase")]
pub enum Severity {
    Error,
    Warning,
    Info,
}

/// 评审问题项
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ReviewIssue {
    pub severity: Severity,
    pub file: String,
    pub line: u32,
    pub message: String,
    pub suggestion: String,
    #[serde(default)]
    pub ignored: bool,
}

/// 改进建议
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ReviewImprovement {
    pub message: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub files: Option<Vec<String>>,
}

/// 代码亮点
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ReviewHighlight {
    pub message: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub files: Option<Vec<String>>,
}

/// AI 评审结果
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AIReviewResult {
    pub id: String,
    pub timestamp: u64,
    pub worktree_path: String,
    pub target_branch: String,
    pub issues: Vec<ReviewIssue>,
    pub improvements: Vec<ReviewImprovement>,
    pub highlights: Vec<ReviewHighlight>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub raw_response: Option<String>,
}

/// AI 评审请求
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AIReviewRequest {
    pub worktree_path: String,
    pub target_branch: String,
    #[serde(default)]
    pub force: bool,
}

/// AI 评审响应
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AIReviewResponse {
    pub success: bool,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub result: Option<AIReviewResult>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub error: Option<String>,
}

/// AI 测试连接响应
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AITestConnectionResponse {
    pub success: bool,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub error: Option<String>,
}

/// AI 评审用的 Diff 统计
#[derive(Debug, Clone)]
pub struct ReviewDiffStats {
    pub additions: u32,
    pub deletions: u32,
    pub changed_files: u32,
}

/// AI 命名建议请求
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AINamingRequest {
    /// 仓库路径
    pub repo_path: String,
    /// 用户输入（可选）
    #[serde(skip_serializing_if = "Option::is_none")]
    pub user_input: Option<String>,
    /// 最近提交数量
    #[serde(default = "default_commit_count")]
    pub commit_count: u32,
}

fn default_commit_count() -> u32 {
    10
}

/// AI 命名建议
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AINamingSuggestion {
    /// 建议的名称
    pub name: String,
    /// 建议类型
    pub suggestion_type: String,
    /// 理由/说明
    pub reason: String,
}

/// AI 命名建议响应
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AINamingResponse {
    pub success: bool,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub suggestions: Option<Vec<AINamingSuggestion>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub error: Option<String>,
}

File: code/src/types/ai.ts
/**
 * AI 评审相关类型定义
 */

/**
 * AI 提供商类型
 */
export type AIProvider = 'openai' | 'claude' | 'ollama' | 'custom';

/**
 * AI 配置
 */
export interface AIConfig {
  /** AI 提供商 */
  provider: AIProvider;
  /** API Key（加密存储） */
  apiKey: string;
  /** API 基础 URL（自定义端点使用） */
  baseUrl?: string;
  /** 模型名称 */
  model: string;
  /** 评审语言 */
  language: 'zh' | 'en';
  /** 是否启用自动评审 */
  autoReview: boolean;
}

/**
 * AI 配置的默认值
 */
export const defaultAIConfig: AIConfig = {
  provider: 'openai',
  apiKey: '',
  model: 'gpt-4o-mini',
  language: 'zh',
  autoReview: false,
};

/**
 * 问题严重程度
 */
export type Severity = 'error' | 'warning' | 'info';

/**
 * 评审问题项
 */
export interface ReviewIssue {
  /** 严重程度 */
  severity: Severity;
  /** 文件路径 */
  file: string;
  /** 行号 */
  line: number;
  /** 问题描述 */
  message: string;
  /** 改进建议 */
  suggestion: string;
  /** 是否被忽略 */
  ignored?: boolean;
}

/**
 * 改进建议
 */
export interface ReviewImprovement {
  /** 建议内容 */
  message: string;
  /** 相关文件（可选） */
  files?: string[];
}

/**
 * 代码亮点
 */
export interface ReviewHighlight {
  /** 亮点描述 */
  message: string;
  /** 相关文件（可选） */
  files?: string[];
}

/**
 * AI 评审结果
 */
export interface AIReviewResult {
  /** 唯一标识 */
  id: string;
  /** 评审时间戳 */
  timestamp: number;
  /** 工作树路径 */
  worktreePath: string;
  /** 目标分支 */
  targetBranch: string;
  /** 潜在问题列表 */
  issues: ReviewIssue[];
  /** 改进建议列表 */
  improvements: ReviewImprovement[];
  /** 亮点列表 */
  highlights: ReviewHighlight[];
  /** 原始响应（调试用） */
  rawResponse?: string;
}

/**
 * AI 评审请求
 */
export interface AIReviewRequest {
  /** 工作树路径 */
  worktreePath: string;
  /** 目标分支 */
  targetBranch: string;
  /** 是否强制重新评审 */
  force?: boolean;
}

/**
 * AI 评审响应
 */
export interface AIReviewResponse {
  /** 是否成功 */
  success: boolean;
  /** 评审结果 */
  result?: AIReviewResult;
  /** 错误信息 */
  error?: string;
}

/**
 * AI 测试连接响应
 */
export interface AITestConnectionResponse {
  /** 是否成功 */
  success: boolean;
  /** 错误信息 */
  error?: string;
}

/**
 * 评审状态
 */
export type ReviewStatus = 'idle' | 'loading' | 'success' | 'error';

/**
 * AI 评审状态（Store 用）
 */
export interface AIReviewState {
  /** 当前评审状态 */
  reviewStatus: ReviewStatus;
  /** 当前评审结果 */
  currentResult: AIReviewResult | null;
  /** 错误信息 */
  error: string | null;
  /** 是否显示配置引导 */
  showConfigGuide: boolean;
}

/**
 * AI 评审用的 Diff 统计信息
 */
export interface ReviewDiffStats {
  /** 新增行数 */
  additions: number;
  /** 删除行数 */
  deletions: number;
  /** 修改文件数 */
  changedFiles: number;
}

/**
 * AI 命名建议请求
 */
export interface AINamingRequest {
  /** 仓库路径 */
  repoPath: string;
  /** 用户输入（可选） */
  userInput?: string;
  /** 最近提交数量 */
  commitCount?: number;
}

/**
 * AI 命名建议
 */
export interface AINamingSuggestion {
  /** 建议的名称 */
  name: string;
  /** 建议类型 */
  suggestionType: string;
  /** 理由/说明 */
  reason: string;
}

/**
 * AI 命名建议响应
 */
export interface AINamingResponse {
  /** 是否成功 */
  success: boolean;
  /** 建议列表 */
  suggestions?: AINamingSuggestion[];
  /** 错误信息 */
  error?: string;
}
